src/Raven.Server/SqlMigration/SqlTable.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using Raven.Server.Config;
+
+namespace Raven.Server.SqlMigration
+{
+    public class SqlTable
+    {
+        public readonly string Name;
+        public readonly List<string> PrimaryKeys;
+        public Dictionary<string, string> ForeignKeys { get; private set; }
+        public readonly string InitialQuery;
+        public readonly List<SqlEmbeddedTable> EmbeddedTables;
+
+        protected readonly string PatchScript;
+        protected readonly IDbConnection Connection;
+        protected readonly RavenConfiguration RavenConfiguration;
```

Comment:
I would suggest renaming this to DatabaseConfiguration so it would be more clear what kind of resource we are dealing here with.